File: cpp/program/Gpu.cpp
#pragma once
#include <iostream>
#include <string>
#include "GraphicProcessingUnit.cpp"
#include "Komponen.cpp"

using namespace std;

class Gpu : public GraphicProcessingUnit, public Komponen{
private:
    int coreCount;
    string vendor;
public:
    Gpu(){}

    Gpu(int coreCount, int vendor, int memoryGB, float clockSpeedGHz, string merk, string nama) : GraphicProcessingUnit(memoryGB, clockSpeedGHz), Komponen(merk, nama){
        this->coreCount = coreCount;
        this->vendor = vendor;
    }

    void setcoreCount(int coreCount){
        this->coreCount = coreCount;
    }
    void setvendor(int vendor){
        this->vendor = vendor;
    }

    int getcoreCount(){
        return this->coreCount;
    }
    string getvendor(){
        return this->vendor;
    }

    ~Gpu(){}
};

File: python/program/Gpu.py
from GraphicProcessingUnit import GraphicProcessingUnit
from Komponen import Komponen

class Gpu(GraphicProcessingUnit, Komponen):
    def __init__(self, core_count=0, vendor='', memory_gb=0, clock_speed_ghz=0.0, merk='', nama=''):
        GraphicProcessingUnit.__init__(self, memory_gb, clock_speed_ghz)
        Komponen.__init__(self, merk, nama)
        self.core_count = core_count
        self.vendor = vendor

    def set_core_count(self, core_count):
        self.core_count = core_count

    def set_vendor(self, vendor):
        self.vendor = vendor

    def get_core_count(self):
        return self.core_count

    def get_vendor(self):
        return self.vendor
